### PR TITLE
[Fix](Status) Handle returned overall Status correctly

### PR DESCRIPTION
**Pull request by wyxxxcat**
## Proposed changes

Issue Number: #28160 

## Further comments

In these files, there are still some similar modifications, but in the destructors and constructors, further discussion and analysis are needed

```
be/src/io/cache/block/cached_remote_file_reader.cpp
be/src/io/fs/broker_file_reader.cpp
be/src/io/fs/broker_file_writer.cpp
be/src/io/fs/buffered_reader.cpp
be/src/io/fs/hdfs_file_reader.cpp
be/src/io/fs/hdfs_file_writer.cpp
be/src/io/fs/s3_file_reader.cpp
be/src/olap/delta_writer.cpp
be/src/olap/delta_writer_v2.cpp
be/src/olap/rowset/beta_rowset_writer.cpp
be/src/runtime/buffer_control_block.cpp
be/src/runtime/jsonb_value.h
```

### CODE REVIEW

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

File: be/src/io/fs/benchmark/fs_benchmark_tool.cpp [hunk 1]
```diff
@@ -115,10 +115,14 @@ int main(int argc, char** argv) {
...
-    static_cast<void>(doris::ThreadPoolBuilder("S3FileUploadThreadPool")
-                              .set_min_threads(num_cores)
-                              .set_max_threads(num_cores)
-                              .build(&s3_file_upload_thread_pool));
+    doris::Status st = doris::ThreadPoolBuilder("S3FileUploadThreadPool")
+                               .set_min_threads(num_cores)
+                               .set_max_threads(num_cores)
+                               .build(&s3_file_upload_thread_pool);
+    if (!st.ok()) {
+        std::cout << "init s3 write buffer pool failed" << std::endl;
```
Comment (zclllyybb) [inline be/src/io/fs/benchmark/fs_benchmark_tool.cpp:123]: change all `std::cout` like this to `std::cerr`

File: be/src/olap/compaction.cpp [hunk 2]
```diff
@@ -79,7 +79,10 @@ bool is_rowset_tidy(std::string& pre_max_key, const RowsetSharedPtr& rhs) {
     // check segment size
     auto* beta_rowset = reinterpret_cast<BetaRowset*>(rhs.get());
     std::vector<size_t> segments_size;
-    static_cast<void>(beta_rowset->get_segments_size(&segments_size));
+    Status status = beta_rowset->get_segments_size(&segments_size);
```
Comment (zclllyybb) [inline be/src/olap/compaction.cpp:82]: for usage like this, you can introduce a new marco like `RETURN_FALSE_IF_ERROR` to deal with them

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

File: be/src/io/fs/benchmark/fs_benchmark_tool.cpp [hunk 3]
```diff
@@ -115,10 +115,14 @@ int main(int argc, char** argv) {
...
-    static_cast<void>(doris::ThreadPoolBuilder("S3FileUploadThreadPool")
-                              .set_min_threads(num_cores)
-                              .set_max_threads(num_cores)
-                              .build(&s3_file_upload_thread_pool));
+    doris::Status st = doris::ThreadPoolBuilder("S3FileUploadThreadPool")
+                               .set_min_threads(num_cores)
+                               .set_max_threads(num_cores)
+                               .build(&s3_file_upload_thread_pool);
+    if (!st.ok()) {
+        std::cerr << "init s3 write buffer pool failed" << std::endl;
```
Comment (zclllyybb) [inline be/src/io/fs/benchmark/fs_benchmark_tool.cpp:123]: change all of them. not only this.

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run p0

**Comment by wyxxxcat** [comment]
run p0 10

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**